### ISSUE TITLE
作者您好，我正在尝试复现您的代码，但是我遇到了一些问题

### ISSUE THREAD

**Issue by xiaosu-beep** [sunxue-yy/SpaGRA#6]
我用BARISTASeq数据集时，得到了非常差的结果，我不知道是什么原因造成的，请问我是否缺失了一些关键步骤呢,很抱歉再次打扰您
import os,csv,re
import pandas as pd
import numpy as np
import scanpy as sc
import matplotlib.pyplot as plt
from sklearn.metrics.cluster import adjusted_rand_score
import warnings
warnings.filterwarnings('ignore')
from SpaGRA.utils import *
from SpaGRA.process import *
from SpaGRA import train_model
from sklearn.metrics import normalized_mutual_info_score
import anndata

adata = sc.read("./Data/BARISTASeq/BARISTASeq_Sun2021Integrating_Slice_2_data.h5ad")
adata.var_names_make_unique()

prefilter_genes(adata, min_cells=3)
sc.pp.highly_variable_genes(adata, flavor="seurat_v3", n_top_genes=1000)
sc.pp.normalize_per_cell(adata)
sc.pp.log1p(adata)
Cal_Spatial_Net(adata, rad_cutoff=60)
adata=train_model.train(adata,k=8,n_epochs=200,method='kmeans')
obs_df = adata.obs.dropna()
ARI = adjusted_rand_score(obs_df['SpaGRA'], obs_df['ground_truth'])
print('Adjusted rand index = %.2f' % ARI)
NMI = normalized_mutual_info_score(obs_df['SpaGRA'], obs_df['ground_truth'])
print('Normalized mutual information = %.2f' % NMI)
sc.pp.neighbors(adata, use_rep='SpaGRA')
sc.tl.umap(adata)
plt.rcParams["figure.figsize"] = (3, 3)
sc.pl.umap(adata, color=["SpaGRA", 'ground_truth'], title=['SpaGRA (ARI=%.2f)' % ARI, 'ground_truth'],save="SpaGRA_umap_BARISTASeq")
# print(adata.obsm['spatial'].shape)
# print(adata.obsm['spatial'])
# Display and save clustering results
plt.rcParams["figure.figsize"] = (3, 3)
sc.pl.spatial(adata, color=["SpaGRA", 'ground_truth'], title=['SpaGRA (ARI=%.2f,NMI=%.2f)' % (ARI,NMI), 'ground_truthr'],spot_size=18,save="SpaGRA_BARISTASeq")

**Comment by xiaosu-beep** [comment]
我过滤掉了标签为outside_VISp的数据进行聚类得到了比较好的结果，这样处理是否符合规范

**Comment by xiaosu-beep** [comment]
我过滤掉了标签为outside_VISp的数据进行聚类得到了比较好的结果，请问这样处理是否可以呢~

**Comment by sunxue-yy** [comment]
是的，我也是这样进行处理的



---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2024年11月27日 11:59 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.***> |
| 主题 | Re: [sunxue-yy/SpaGRA] 作者您好，我正在尝试复现您的代码，但是我遇到了一些问题 (Issue #6) |

我过滤掉了标签为outside_VISp的数据进行聚类得到了比较好的结果，请问这样处理是否可以呢~

**Comment by xiaosu-beep** [comment]
感谢您的回复~
